### DISCUSSION TITLE
鼠须管主题方案中的 `spacing` 表现奇怪。

### DISCUSSION THREAD

**Discussion by pdog18** [rime/home#1272] [Q&A] [answered]
`spacing` 这个值，表示的是`preeditLineSpacing`。
在 `inline_preedit = true`时无效(符合预期)。
但是在 `inline_preedit = false` 时，似乎有错误的表现? 它的值是正确的，但是从 UI 上观察到d的效果有些怪异:
<img width="176" alt="WeChat1753ce23754b29f1df49ec2ce757f82c" src="https://user-images.githubusercontent.com/17060376/233847458-d41ce76a-2568-456e-88e9-dd017bd7451e.png">
(红色区域是 `spacing` 的值，表现正确，但是黄色区域的背景颜色表现怪异)

这大概也不是 spacing 的问题，而是首个候选词选中时的高亮背景区域的问题?

**Comment by pdog18** [comment]
~~大概理解会产生这种怪异现象的原因是:「preedit 区域，有可能存在 1 行以上的文字」。
这个 spacing 的作用，不是专门用来间隔 「preedig 与候选词区」的。
但是有这种表现依旧有些奇怪。~~

**Reply by LEOYoon-Tsaw** [reply]
它就是「专门用来间隔 『preedig 与候选词区』的。」

**Reply by pdog18** [reply]
> 它就是「专门用来间隔 『preedig 与候选词区』的。」

这里是我搞错了! 
把 line_spacing 产生的效果，认做是 spacing 产生的效果了。

**Comment by LEOYoon-Tsaw** [comment]
這個spacing指的是preedit區和candidate區之間的間隔。你設了一個很大的值，當然間隔就大得離譜，有何奇怪？

**Reply by pdog18** [reply]
> 這個spacing指的是preedit區和candidate區之間的間隔。你設了一個很大的值，當然間隔就大得離譜，有何奇怪？

「大的离谱」指得是图中红色区域，这是符合预期的，但是图中黄色区域，有高亮背景颜色(这一点是令我感到奇怪的)。(难道设计就是如此吗?)

**Reply by LEOYoon-Tsaw** [reply, marked as answer]
candidate區和preedit區平分這個空間是最優雅的

**Reply by pdog18** [reply]
> candidate區和preedit區平分這個空間是最優雅的

所以这不是 bug，而是专门设计的 feature

**Reply by LEOYoon-Tsaw** [reply]
當然也可以增加更多配置項來單獨控制上下兩部分的大小，只是我覺得為了這麼點東西弄太多配置項不太有必要

**Reply by pdog18** [reply]
> 當然也可以增加更多配置項來單獨控制上下兩部分的大小，只是我覺得為了這麼點東西弄太多配置項不太有必要

如果可以自定义 logo 在 theme 上，可能有些用家会很喜欢!(不包括我!)。

我觉得这个功能是非常有趣的，并且对吸引有这癖好(?想不到别的词)的用户来说非常非常非常有趣。(哈哈哈哈)

**Reply by pdog18** [reply]
> > 當然也可以增加更多配置項來單獨控制上下兩部分的大小，只是我覺得為了這麼點東西弄太多配置項不太有必要
> 
> 如果可以自定义 logo 在 theme 上，可能有些用家会很喜欢!(不包括我!)。
> 
> 我觉得这个功能是非常有趣的，并且对吸引有这癖好(?想不到别的词)的用户来说非常非常非常有趣。(哈哈哈哈)

我甚至在这一会儿想到，视频网站上，一些视频创作者，在使用带有自己设计的 logo 的输入法，飞快的键入文字。
这时候弹幕/评论发现了，询问「请问这是哪款输入法」时，我可以特别热情的告诉他们是 Rime(叉腰)
